src/services/specialtyService.ts

```diff
@@ -1,13 +1,50 @@
-import { specialties } from '../mockData/mockData';
-import { Specialty } from '../typeDefs/gqlTypes';
+
+import { Specialty, SpecialtyName } from '../typeDefs/gqlTypes';
+import {Specialty as PrismaSpecialty, SpecialtyName as PrismaSpecialtyName} from '@prisma/client';
+import prisma from '../db/client';
+
+type SpecialtyAndNames = (PrismaSpecialty & { names: PrismaSpecialtyName[]})
+
+function convertPrismaToGraphQLSpecialty(input: SpecialtyAndNames) {
+    const ret = {
+        id: String(input.id),
+        names: Array<SpecialtyName>()
+    } as Specialty;
+
+    for (let i = 0; i < input.names.length; i++) {
+        ret.names?.push({
+            name: input.names[i].name,
+            locale: input.names[i].locale
+        });
+    }
+
+    return ret;
+}
 
 // TODO: add a validation step for incoming parameters
-export const getSpecialtyById = (id: string) => {
-    const matchingSpecialty = specialties.find(
-        (medicalField: Specialty) => medicalField.id === id
-    );
+export const getSpecialtyById = async (id: string) => {
+    const found = await prisma.specialty.findUnique({ where: {
+        id: parseInt(id)
+    },
+    include: {
+        names: true
+    }});
 
-    return matchingSpecialty;
+    if (found) {
```

Comment:
readability recommendation: I recommend using the "[early return pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8)"

Reply:
I think that's a great pattern when there are nested ifs -- to break the logic out of nests.

But in this case, seems a bit "six of one, half-dozen of the other" 🤔

Reply:
Anyway it's about to disappear as part of another refactor